### PR TITLE
[CALCITE-5169] 'xx < 1 OR xx > 1' cannot be simplified to 'xx <> 1'

### PR DESCRIPTION
**Pull request by libenchao**

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/rex/RexProgramTest.java [hunk 1]
```diff
@@ -1443,11 +1443,15 @@ private void checkExponentialCnf(int n) {
...
 
     // "a = 1 or a <> 2" could (and should) be simplified to "a <> 2"
-    // but can't do that right now
     checkSimplifyFilter(
         or(eq(aRef, literal1),
             ne(aRef, literal2)),
-        "OR(=(?0.a, 1), <>(?0.a, 2))");
+        "<>(?0.a, 2)");
+
+    checkSimplifyFilter(
```
Comment (rkondakov) [inline core/src/test/java/org/apache/calcite/rex/RexProgramTest.java:1451]: What about adding a clarifying comment as we have in other tests? I.e. `a < 1 or a > 1 could be simplified to a <> 1`

Comment (libenchao) [reply core/src/test/java/org/apache/calcite/rex/RexProgramTest.java:1452]: good idea. done.

**Comment by rkondakov** [comment]
Valuable improvement! LGTM

**Review by NobiGo — APPROVED** [review]
LGTM

**Review by chunweilei — COMMENTED** [review]
LGTM

**Comment by chunweilei** [comment]
Could you squash commits into a single one? @libenchao

**Comment by libenchao** [comment]
@chunweilei Done. Thanks for the review.

**Review by chunweilei — APPROVED** [review]
+1. Will merge shortly.

**Pull request closed without merging**